File: src/main/java/com/brenoepic/clothes/ClothingManager.java
package com.brenoepic.clothes;

import com.eu.habbo.Emulator;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.*;
import java.util.stream.Collectors;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class ClothingManager {
  private final HashMap<Integer, Clothing> clothes;

  public ClothingManager() {
    this.clothes = new HashMap<>();
    this.load();
  }

  public boolean load() {
    this.dispose();
    try (final Connection connection = Emulator.getDatabase().getDataSource().getConnection();
        final PreparedStatement statement =
            connection.prepareStatement("SELECT * FROM `special_looks`");
        final ResultSet set = statement.executeQuery()) {
      while (set.next()) {
        final Clothing cloth = new Clothing(set);
        this.clothes.put(set.getInt("id"), cloth);
      }
    } catch (SQLException e) {
      log.error("[NFTAvatarPlugin]", e);
      return false;
    }
    log.info("[NFTAvatarPlugin] Loaded {} NFT avatars successfully!", this.clothes.size());
    return true;
  }

  public void dispose() {
    this.clothes.clear();
  }

  public Clothing getClothingBySetId(final int setId) {
    return this.clothes.values().stream()
        .filter(clothing -> filterSetId(setId, clothing))
        .findFirst()
        .orElse(null);
  }

  private static boolean filterSetId(int setId, Clothing clothing) {
    return clothing.getSet() != null
        && clothing.getType().equals("setid")
        && Integer.parseInt(clothing.getSet()) == setId;
  }

  public Clothing getClothing(final String figure) {
    return this.clothes.values().stream()
        .filter(clothing -> filterFigure(figure, clothing))
        .findFirst()
        .orElse(null);
  }

  private static boolean filterFigure(String figure, Clothing clothing) {
    return clothing.getSet() != null
        && clothing.getType().equals("figure")
        && clothing.getSet().equals(figure);
  }

  public List<Integer> getEffectList() {
    return this.clothes.values().stream().map(Clothing::getEffect).collect(Collectors.toList());
  }
}

File: src/main/java/io/github/brenoepics/clothes/ClothingManager.kt
package io.github.brenoepics.clothes

import com.eu.habbo.Emulator
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import java.sql.SQLException
import java.util.stream.Collectors

class ClothingManager {
    private val clothes = HashMap<Int, Clothing?>()

    init {
        this.load()
    }

    fun load(): Boolean {
        this.dispose()
        try {
            Emulator.getDatabase().dataSource.connection.use { connection ->
                connection.prepareStatement("SELECT * FROM `special_looks`").use { statement ->
                    statement.executeQuery().use { set ->
                        while (set.next()) {
                            clothes[set.getInt("id")] = Clothing(set)
                        }
                    }
                }
            }
        } catch (e: SQLException) {
            log.error("[NFTAvatarPlugin]", e)
            return false
        }
        log.info(
            "[NFTAvatarPlugin] Loaded {} NFT avatars successfully!",
            clothes.size
        )
        return true
    }

    fun dispose() {
        clothes.clear()
    }

    fun getClothingBySetId(setId: Int): Clothing? {
        return clothes.values.stream()
            .filter { clothing: Clothing? -> filterSetId(setId, clothing) }
            .findFirst()
            .orElse(null)
    }

    fun getClothing(figure: String): Clothing? {
        return clothes.values.stream()
            .filter { clothing: Clothing? -> filterFigure(figure, clothing) }
            .findFirst()
            .orElse(null)
    }

    val effectList: List<Int>
        get() = clothes.values.stream().map { obj: Clothing? -> obj!!.effect }.collect(Collectors.toList())

    companion object {
        private val log: Logger = LoggerFactory.getLogger(ClothingManager::class.java)
        private fun filterSetId(setId: Int, clothing: Clothing?): Boolean {
            return clothing!!.set != null && clothing.type == "setid" && clothing.set!!.toInt() == setId
        }

        private fun filterFigure(figure: String, clothing: Clothing?): Boolean {
            return clothing!!.set != null && clothing.type == "figure" && clothing.set == figure
        }
    }
}
